### PR TITLE
[Not Yet] Swagger

### PR DESCRIPTION
**Pull request by kyuksel**
It needs testing.

### CODE REVIEW

File: src/main/resources/swagger/api-docs.yaml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+            - profile
+
+##########################################################################################
+## DEFINITIONS
+##########################################################################################
+definitions:
+  Project:
+    description: ''
+    required:
+      -projectName
```
Comment (kyuksel) [inline src/main/resources/swagger/api-docs.yaml]: @helgridly I get the error below pointing to the line above. Can you tell why?

image

Comment (helgridly) [reply src/main/resources/swagger/api-docs.yaml]: You might just need a space between the dash and projectName

Comment (kyuksel) [reply src/main/resources/swagger/api-docs.yaml]: Thank you!

**Comment by helgridly** [comment]
There's more you're missing here -- a whole folder of Swagger resources to dump in /resources, plus config and a bit of work to set up the Swagger routes. See Leo for an example:

https://github.com/DataBiosphere/leonardo/blob/develop/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/Boot.scala
https://github.com/DataBiosphere/leonardo/blob/develop/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/SwaggerRoutes.scala

File: src/main/resources/swagger/api-docs.yaml (hunk 1 above)
Comment (rushtong) [reply src/main/resources/swagger/api-docs.yaml]: As a side note, see this comment for swagger validation resources: https://broadinstitute.atlassian.net/browse/DSPTO-246?focusedCommentId=278478&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-278478

In practice, I found that you need three layers of validation to catch all potential errors:
* Intellij's swagger validation - good first pass
* Swagger's version of validation: https://editor.swagger.io/ - good second pass
* Both of those can miss some yaml formatting errors, so you need a third pass with https://codebeautify.org/yaml-validator

Comment (kyuksel) [reply src/main/resources/swagger/api-docs.yaml]: Thanks @rushtong!

File: configs/gpalloc.conf.ctmpl [hunk 2]
```diff
@@ -27,4 +27,9 @@ liquibase {
   changelog = "org/broadinstitute/dsde/workbench/gpalloc/liquibase/changelog.xml"
   initWithLiquibase = true
 }
+
+swagger {
+  googleClientId = "806222273987-fne0mu7nnof7ifg6e0uj6kqsdmk0tck2.apps.googleusercontent.com"
+  realm = "broad-dsde-local"
+}
```
Comment (kyuksel) [inline configs/gpalloc.conf.ctmpl]: @helgridly I can't imagine above is what we want but I didn't know what else to insert.

**Pull request merged**